### PR TITLE
[ci/multinode] Build multinode image with OpenSSH before running tests

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Currently we install OpenSSH on the fly in fake multinode docker testing. Instead we can speed testing up a fair bit by building a Docker image which includes OpenSSH first and then run tests with this image.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by richardliaw — APPROVED** [review]

File: python/ray/autoscaler/_private/fake_multi_node/test_utils.py [hunk 1]
```diff
@@ -134,7 +135,16 @@ def update_config(self, config: Optional[Dict[str, Any]] = None):
...
             # No image specified, trying to parse from buildkite
-            self._docker_image = os.environ.get("RAY_DOCKER_IMAGE", None)
+            docker_image = os.environ.get("RAY_DOCKER_IMAGE", None)
+
+            if not docker_image:
+                # If still no docker image, use one according to Python version
+                mj = sys.version_info.major
+                mi = sys.version_info.minor
+
+                docker_image = f"rayproject/ray:nightly-py{mj}{mi}-cpu"
```
Comment (richardliaw) [inline python/ray/autoscaler/_private/fake_multi_node/test_utils.py:145]: can we bump this to a global constant?

File: doc/source/fake-autoscaler.rst [hunk 2]
```diff
@@ -194,11 +194,17 @@ For the Ray OSS Buildkite environment, we thus set some environment variables:
   variable.
 
 Lastly, docker-compose obviously requires a docker image. The default docker image is ``rayproject/ray:nightly``.
-However, on our Ray OSS Buildkite environment, we want to test changes from the respective PR or Branch. Thus, we set
+The docker image requires ``openssh-server`` to be installed and enabled. In Buildkite we build a new image from
+``rayproject/ray:nightly-py37-cpu`` to avoid installing this on the fly for every node (which is the default way).
```
Comment (xwjiang2010) [inline doc/source/fake-autoscaler.rst:198]: for all supported python version?

**Review by xwjiang2010 — APPROVED** [review]
Great! Thanks Kai. Just one minor comment.

File: doc/source/fake-autoscaler.rst (hunk 2 above)
Comment (krfricke) [reply doc/source/fake-autoscaler.rst:198]: Multinode tests on buildkite only run on Python 3.7 (as do most Tune tests). The build script supports all base images though.

**Pull request merged**